materialize-snowflake/config.go

```diff
@@ -0,0 +1,293 @@
+	Account   string `json:"account" jsonschema:"title=Account,description=The Snowflake account identifier." jsonschema_extras:"order=1"`
+	User      string `json:"user" jsonschema:"-"`
+	Password  string `json:"password" jsonschema:"-"`
+	Database  string `json:"database" jsonschema:"title=Database,description=The SQL database to connect to." jsonschema_extras:"order=4"`
+	Schema    string `json:"schema" jsonschema:"title=Schema,description=Database schema for bound collection tables (unless overridden within the binding resource configuration) as well as associated materialization metadata tables." jsonschema_extras:"order=5"`
+	Warehouse string `json:"warehouse,omitempty" jsonschema:"title=Warehouse,description=The Snowflake virtual warehouse used to execute queries. Uses the default warehouse for the Snowflake user if left blank." jsonschema_extras:"order=6"`
+	Role      string `json:"role,omitempty" jsonschema:"title=Role,description=The user role used to perform actions." jsonschema_extras:"order=7"`
+
+	Credentials credentialConfig `json:"credentials" jsonschema:"title=Authentication"`
+
+	Advanced advancedConfig `json:"advanced,omitempty" jsonschema:"title=Advanced Options,description=Options for advanced users. You should not typically need to modify these." jsonschema_extras:"advanced=true"`
+}
+
+type advancedConfig struct {
+	UpdateDelay string `json:"updateDelay,omitempty" jsonschema:"title=Update Delay,description=Potentially reduce active warehouse time by increasing the delay between updates. Defaults to 30 minutes if unset.,enum=0s,enum=15m,enum=30m,enum=1h,enum=2h,enum=4h"`
+}
+
+// ToURI converts the Config to a DSN string.
+func (c *config) ToURI(tenant string) string {
+	// Build a DSN connection string.
+	var configCopy = c.asSnowflakeConfig(tenant)
+	// client_session_keep_alive causes the driver to issue a periodic keepalive request.
+	// Without this, the authentication token will expire after 4 hours of inactivity.
+	// The Params map will not have been initialized if the endpoint config didn't specify
+	// it, so we check and initialize here if needed.
+	if configCopy.Params == nil {
+		configCopy.Params = make(map[string]*string)
+	}
+	configCopy.Params["client_session_keep_alive"] = &trueString
+	dsn, err := sf.DSN(&configCopy)
+	if err != nil {
+		panic(fmt.Errorf("building snowflake dsn: %w", err))
+	}
+
+	return dsn
+}
+
+func (c *config) privateKey() (*rsa.PrivateKey, error) {
+	if c.Credentials.AuthType == JWT {
+		// When providing the PEM file in a JSON file, newlines can't be specified unless
+		// escaped, so here we allow a escape hatch to parse these PEM files
+		var pkString = strings.ReplaceAll(c.Credentials.PrivateKey, "\\n", "\n")
```

Comment:
I have checked that in the UI when we submit a multi-line secret through the upload form, newlines are present as `\n` in the string, so this should work with the UI form